### PR TITLE
Added Audio Steganography

### PR DESCRIPTION
**Pull request by MayankkumarTank**
## Description

Here, you can see two different scripts 1. endcoder.py and 2. decoder.py. First, talking about the encoder.py. This script is used to hide the data in the audio by the LSB algorithm. The first user is requested to enter the file name and then the secret message after that script will generate an encoded audio file. Second, decoder.py is used to decode the msg from the encoded audio file. In this script, the user is requested to enter the name of the audio file and the script will automatically decode the secret msg in it. Here, I attached one sample audio file iphone.mp3.

Fixes: #790

### CODE REVIEW

**Comment by MayankkumarTank** [comment]
Hello sir, I am getting an error in the Lint code base. It says (please find attachment),

image

As per my understanding, There is an error with setparams and writeframes but when I am running my code in my system. I am not getting any errors. Please can you help me out?

**Comment by MayankkumarTank** [comment]
Hello sir, How can I link this PR to issue-790? 

(By mistake, I closed this PR but I reopened it in a minute. So, sorry for that and looking forward to getting merged).

Thank you

**Comment by kaustubhgupta** [comment]
> Hello sir, How can I link this PR to issue-790?
> 
> (By mistake, I closed this PR but I reopened it in a minute. So, sorry for that and looking forward to getting merged).
> 
> Thank you

I have done that

File: Audio Steganography/encoder.py [hunk 1]
```diff
@@ -0,0 +1,50 @@
+import wave
+from os import path
+from pydub import AudioSegment
+
+#asking user to enter file name
+file_name_mp3 = input("Enter the .mp3 file name : ")
+destination = "generated_wav.wav"
```
Comment (kaustubhgupta) [inline Audio Steganography/encoder.py:7]:
```suggestion
destination = "./Audio Steganographygenerated_wav.wav"
```

File: Audio Steganography/encoder.py [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+#convert secret message to corresponding bits.
+sb = list(map(int, ''.join([bin(ord(c)).lstrip('0b').rjust(8,'0') for c in secret_message])))
+
+#replacing the last bit of all frames with secret message bit
+for index,c_bit in enumerate(sb):
+    audio_byte_arr[index] = (audio_byte_arr[index] & 254) | c_bit
+
+#wrinting into .wav file
+f = wave.open('modified.wav', 'wb')
```
Comment (kaustubhgupta) [inline Audio Steganography/encoder.py:42]: Save the encrypted file with the same file name as entered by the user encrypted word in the bracket. Example: 

input: audio.mp3
output_file_name: audio (encrypted).mp3

**Review by kaustubhgupta — CHANGES_REQUESTED** [review]
Delete iphone.mp3 and modified.wav along with these changes:

File: Audio Steganography/encoder.py (hunk 2 above)
Comment (MayankkumarTank) [reply Audio Steganography/encoder.py]: Hello sir,
          I made all the required changes. Please check it.

Thank you.

**Comment by kaustubhgupta** [comment]
I am not able to test the script locally, @antrikshmisri @santushtisharma10 can you confirm the working of the script?

File: Audio Steganography/encoder.py [hunk 3]
```diff
@@ -0,0 +1,51 @@
+import wave
+from os import path
+from pydub import AudioSegment
+
+#asking user to enter file name
+file_name_mp3 = input("Enter the .mp3 file name : ")
+destination = "./Audio_Steganographygenerated_wav.wav"
+
```
Comment (santushtisharma10) [inline Audio Steganography/encoder.py:8]: @MayankkumarTank  take the input of destination from user

**Review by santushtisharma10 — CHANGES_REQUESTED** [review]
this is the hidden message being shown
image

fix this

**Comment by kaustubhgupta** [comment]
@MayankkumarTank Did you update the changes suggested by @santushtisharma10?

**Comment by kaustubhgupta** [comment]
@MayankkumarTank will you update the PR?

**Comment by MayankkumarTank** [comment]
Sorry sir for no input from my side.

Due to health issue, i am not able to work. I was in complete bed rest. I will update it soon.

Once again sorry and Thank you.

**Comment by kaustubhgupta** [comment]
> Sorry sir for no input from my side.
> 
> Due to health issue, i am not able to work. I was in complete bed rest. I will update it soon.
> 
> Once again sorry and Thank you.

oh okay, You should have informed me in the first place. Take your time, 3-4 days or till you feel better.

**Comment by kaustubhgupta** [comment]
@MayankkumarTank are you in the state of completing this issue? If no, then I can assign it to someone else.

**Comment by kaustubhgupta** [comment]
PR closed and issue unassigned due to participant's health

**Pull request closed without merging**